src/game/game.cpp

```diff
@@ -89,4 +83,17 @@ namespace game
 
 		return count;
 	}
+
+	class component final : public component_interface
+	{
+	public:
+		void post_unpack() override
+		{
+			current = reinterpret_cast<const char*>(0xC2F028) == "multiplayer"s
+				? gamemode::multiplayer 
+				: gamemode::zombies;
+		}
+	};
```

Comment:
Game isn't supposed to be a component and using it like this is wrong because it's supposed to be initialized before all the other components. If you really wanted to make it a component you should use a different function from `post_unpack` and call it before it. Either way `game::current` can be initialized without the need of a function, so remove this stuff.